### DISCUSSION TITLE
CI/CD for a newbie

### DISCUSSION THREAD

**Discussion by mbkcode** [community/community#124065] [Actions]
### Select Topic Area

Question

### Body

As a beginner, how can i effectively use github ci/cd tools

**Comment by aakash414** [comment]
Here are some tips for effectively using GitHub CI/CD tools as a beginner:
1. Start with a pre-built workflow.
There are many pre-built workflows available in the GitHub Marketplace, so you don't need to start from scratch. You can find a workflow that matches your project's needs and customize it as needed.
2. Use the workflow visualizer.
The workflow visualizer is a great tool for understanding how your workflow works. It shows you the different steps in your workflow and how they are connected.
3. Use live logs.
Live logs allow you to see the output of your workflow as it runs. This can be helpful for debugging problems.
4. Trigger your workflow manually.
You can trigger your workflow manually to test it or to run it on demand.
5. Use a CI/CD best practices.
There are a number of best practices that you can follow to make your CI/CD pipeline more effective. These best practices include committing early and often, keeping your builds green, and building only once.

**Comment by ebndev** [comment]
Hi @mbkcode & @aakash414, thanks for being a part of the GitHub Community!

Please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
